src/h5web/test-utils.tsx

```diff
@@ -59,3 +75,163 @@ export function prepareForConsoleError() {
   };
 }
 /* eslint-enable no-console */
+
+type Opts = Partial<Pick<MyHDF5Dataset, 'id' | 'parents' | 'attributes'>>;
+
+export function makeMyGroup(
+  name: string,
+  children: MyHDF5Entity[] = [],
+  opts: Opts = {}
+): MyHDF5Group {
+  const { id = name, parents = [], attributes = [] } = opts;
+
+  const group: MyHDF5Group = {
+    uid: nanoid(),
+    id,
+    name,
+    kind: MyHDF5EntityKind.Group,
+    parents,
+    children,
+    attributes,
+  };
+
+  group.parents.forEach((parent) => {
+    parent.children = [...parent.children, group];
+  });
+
+  group.children.forEach((child) => {
+    child.parents = [group, ...child.parents];
+  });
+
+  return group;
+}
+
+export function makeMyDataset<S extends HDF5Shape, T extends HDF5Type>(
+  name: string,
+  shape: S,
+  type: T,
+  opts: Opts = {}
+): MyHDF5Dataset<S, T> {
+  const { id = name, parents = [], attributes = [] } = opts;
+
+  const dataset: MyHDF5Dataset<S, T> = {
+    uid: nanoid(),
+    id,
+    name,
+    kind: MyHDF5EntityKind.Dataset,
+    parents,
+    attributes,
+    shape,
+    type,
+  };
+
+  dataset.parents.forEach((parent) => {
+    parent.children = [...parent.children, dataset];
+  });
+
+  return dataset;
+}
+
+export function makeMySimpleDataset<T extends HDF5Type>(
+  name: string,
+  type: T,
+  dims: HDF5Dims,
+  opts: Opts = {}
+): MyHDF5Dataset<HDF5SimpleShape, T> {
+  return makeMyDataset(name, makeSimpleShape(dims), type, opts);
+}
+
+export function makeMyDatatype<T extends HDF5Type>(
+  name: string,
+  type: T,
+  opts: Opts = {}
+): MyHDF5Datatype<T> {
+  const { id = name, parents = [], attributes = [] } = opts;
+
+  const datatype: MyHDF5Datatype<T> = {
+    uid: nanoid(),
+    id,
+    name,
+    kind: MyHDF5EntityKind.Datatype,
+    parents,
+    attributes,
+    type,
+  };
+
+  datatype.parents.forEach((parent) => {
+    parent.children = [...parent.children, datatype];
+  });
+
+  return datatype;
+}
+
+export function withMyAttributes<T extends MyHDF5Entity>(
+  entity: T,
+  attributes: HDF5Attribute[]
+): T {
+  return {
+    ...entity,
+    attributes: [...entity.attributes, ...attributes],
+  };
+}
+
+export function withMyInterpretation<
+  T extends MyHDF5Dataset<HDF5SimpleShape, HDF5NumericType>
+>(dataset: T, interpretation: NxInterpretation): T {
+  return withMyAttributes(dataset, [
+    makeStrAttr('interpretation', interpretation),
+  ]);
+}
```

Comment:
It's so I don't have to recreate entities from scratch in the tests, when all I need is the same entity with an extra attribute. For instance, `datasetInt1D` and `spectrumDatasetInt1D`:

```ts
const datasetInt1D = makeMySimpleDataset('dataset_int_1d', intType, [5]);
const spectrumDatasetInt1D = withMyInterpretation(
  datasetInt1D,
  NxInterpretation.Spectrum
);
```

Instead of:

```ts
const spectrumDatasetInt1D = makeMySimpleDataset('dataset_int_1d', intType, [5]);
const datasetInt1D = makeMySimpleDataset('dataset_int_1d', intType, [5], {
  attributes: [makeStrAttr('interpretation', NxInterpretation.Spectrum)],
});
```

`withMyInterpretation` is more readable and provides a stricter API by requiring a value of type `NxInterpretation`. I could have created a function called `makeSimpleMyDatasetWithInterpretation`, but it seemed unnecessarily complicated.

Reply:
All right and you are doing so by creating a new entity. I misread and thought you were mutating the original entity :scream: 

All good !